tests/rptest/tests/tx_admin_api_test.py

```diff
@@ -243,3 +244,56 @@ def test_all_transactions(self):
                 expected_partitions[topic.name]) == topic.partition_count
             for partition in range(topic.partition_count):
                 assert partition in expected_partitions[topic.name]
+
+    @cluster(num_nodes=3)
+    def test_delete_topic_from_tx(self):
+        tx_id = "0"
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': tx_id,
+        })
+        producer.init_transactions()
+        producer.begin_transaction()
+
+        for topic in self.topics:
+            for partition in range(topic.partition_count):
+                producer.produce(topic.name, '0', '0', partition)
+
+        producer.flush()
+
+        txs_info = self.admin.get_all_transactions()
+        assert len(txs_info) == 1
```

Comment:
use `assert condition, f"..."` to add context to the failed assertion to make it easier to debug if the assertion fails.